src/api/stats/common/getCompoundV2Apys.ts

```diff
@@ -0,0 +1,214 @@
+import BigNumber, { BigNumber as BigNumberStatic } from 'bignumber.js';
+import Web3 from 'web3';
+import { AbiItem } from 'web3-utils';
+import { ChainId } from '../../../../packages/address-book/address-book';
+import { getTotalPerformanceFeeForVault } from '../../vaults/getVaultFees';
+import getBlockTime from '../../../utils/getBlockTime';
+import fetchPrice from '../../../utils/fetchPrice';
+import { compound } from '../../../utils/compound';
+import { getContractWithProvider } from '../../../utils/contractHelper';
+import { BASE_HPY } from '../../../constants';
+
+const IToken = require('../../../abis/VToken.json');
+const IComptroller: AbiItem[] = require('../../../abis/IComptroller.json');
+
+const SECONDS_PER_YEAR = 31536000;
+
+const getCompoundV2ApyData = async (params: CompoundV2ApysParams) => {
+  params.compOracle = params.compOracle || 'tokens';
+  params.compDecimals = params.compDecimals || '1e18';
+  params.comptrollerAbi = params.comptrollerAbi || IComptroller;
+  params.cTokenAbi = params.cTokenAbi || IToken;
+
+  const [calculatedBlockTime] = await Promise.all([getBlockTime(params.chainId)]);
+  params.secondsPerBlock = params.secondsPerBlock || calculatedBlockTime;
+
+  let apys = {};
+
+  let promises = [];
+  params.pools.forEach(pool => promises.push(getPoolApy(params, pool)));
```

Comment:
would be super cool if its rewritten in a multicall way (check `common/getMasterChefApy`).
could be supplyCalls and borrowCalls with 1 multicall total vs current 40+

Reply:
I started there, didn't get it working, then came at it from a different direction. Now that the calls are right, I can bring it back to multicall.